zebra/if_netlink.c

```diff
@@ -350,11 +350,7 @@ static void netlink_vrf_change(struct nlmsghdr *h, struct rtattr *tb,
 			}
 		}
 
-		/*
-		 * vrf_get is implied creation if it does not exist
-		 */
-		vrf = vrf_get((vrf_id_t)ifi->ifi_index,
-			      name); // It would create vrf
+		vrf = vrf_update((vrf_id_t)ifi->ifi_index, name);
```

Comment:
umm - this doesn't address the question I had. you were calling 'vrf_disable' directly here, and you've just moved that call into 'vrf_update'. that doesn't resolve the question about which pthread is running - it just moves the problem?

Reply:
mark abstracting it into vrf_update() was my request to get the logic out of netlink code and into the lib/vrf.c code.

I'm not sure I fully understand what you were requesting here either. Do you have some other code doing what you want you can point him to?

Reply:
oh, sorry - maybe it's my bad then - I thought we wanted something more elaborate to happen, but if you're ok with the flow here now, then I'm ok too

Reply:
ideally all this should probably come from the dplane pthread one day but its out of the scope of this PR imo

Reply:
yes, absolutely, this PR isn't responsible for that

Reply:
Thanks @mjstapp , @sworleys , can you please approve this PR ?